pymc3/model.py

```diff
@@ -1518,3 +1472,315 @@ def all_continuous(vars):
+                    yield parent
+
+
+def get_sub_dag(dag, input_nodes, force=True):
+    """Get a new DiGraph instance which is like a right outer join
+    of `dag` with a list of input nodes provided in `input_nodes`.
+    What this means is that it will look for the `input_nodes` inside
+    `dag`, the nodes which are contained in `dag`, along with all their
+    predecessors will be used to get a subgraph of `dag`. Then, the
+    remaining nodes will be added to the `DiGraph` with
+    `add_to_dependence_dag`. Finally, this instance is returned.
+    In summary, it copies the shared part of `dag`, given the nodes in
+    `input_nodes`, and then adds onto that.
+
+    Parameters
+    ----------
+    dag : DiGraph instance
+        Dependence DAG DiGraph on which to perform the right outer join
+        operation.
+    input_nodes : list or scalar
+        If it is a scalar `input_nodes` will be converted to a list as
+        `[input_nodes]`. `input_nodes` is a list of nodes that will be
+        used to create a new `DependenceDAG` instance. The part of the DAG
+        that is shared with `self`, will be copied, and the rest will be
+        added.
+    force : bool (optional)
+        If True, the nodes that must be added, will be added with the
+        force flag set to True. [Default is True]
+
+    Returns
+    -------
+    A tuple. The first element is a new `DiGraph` instance that results from
+    the right outer join operation.
+    The second element in the output tuple is the dictionary of indices to
+    nodes `{index: node}`. Each key will be the position of the node provided
+    in `input_nodes` and the value will be the node that was added to the
+    returned `DiGraph` instance, which could either be a
+    `WrapAsHashable` instance or `input_nodes[index]` itself.
+    """
+    if not isinstance(dag, networkx.DiGraph):
+        raise TypeError('Input `dag` must be an instance of networkx.DiGraph. '
+                        'Got {}, which has type {}, instead.'.
+                        format(dag, type(dag)))
+    if not isinstance(input_nodes, list):
+        input_nodes = [input_nodes]
+    index = {}
+    copied_bunch = set()
+    nodes_to_add = []
+    for i, node in enumerate(input_nodes):
+        if node in dag:
+            index[i] = node
+            copied_bunch.add(node)
+            copied_bunch.update(networkx.ancestors(dag, node))
+        else:
+            nodes_to_add.append((i, node))
+
+    subgraph = dag.subgraph(copied_bunch).copy()
+    for node_index, node in nodes_to_add:
+        added_node = add_to_dependence_dag(subgraph,
+                                           node,
+                                           force=force)
+        if node_index is not None:
+            index[node_index] = added_node
+    return subgraph, index
+
+
+def matching_dependence_dags(a, b):
+    """A helper function inteaded to be used during debugging. Is true is
```

Comment:
`intended`

Reply:
Oops, I'll fix the typo but first I'll wait to see what is decided about the dag before pushing it.